src/mscorlib/shared/System/Globalization/CompareInfo.cs

```diff
@@ -620,9 +620,14 @@ public virtual bool IsPrefix(string source, string prefix, CompareOptions option
                     SR.ArgumentNull_String);
             }
 
+            return IsPrefix(source.AsReadOnlySpan(), prefix.AsReadOnlySpan(), options);
```

Comment:
IsPrefix method is public virtual. What happens if somebody overrides it?

Reply:
I believe Jan is highlighting that if someone is currently overriding it, and we switch code to use the new span-based IsPrefix method instead of the string-based IsPrefix method, we'll be bypassing the existing overrides.

Reply:
The span-based IsPrefix is an internal implementation detail of the string IsPrefix. How will the existing string IsPrefix overrides be affected by this change? Essentially, I just extracted the code from the string method into an internal span based method.

Reply:
> How will the existing string IsPrefix overrides be affected by this change?

The new ones won't respect any existing overrides already in place.

A similar situation has arisen in streams in the past.  Consider MemoryStream.  Imagine you'd derived a MaxSizeMemoryStream : MemoryStream, where you overrode Write(byte[], int, int) to track how much data has been written and to throw an exception from Write if too much data is written.  You've overridden all possible Write methods.  Now in the next release we add a new virtual Write(Span) method, and code starts using it with your type.  There's now a way to Write to this stream that doesn't respect the overrides and wishes of your type.

I don't know if it's something that needs to be addressed here; I'm just calling out a potential issue.

Reply:
Ah I see! Thanks for the explanation. 

I understand that if we add new public span overloads and there are implicit conversions, then existing overrides won't be called. However, `public virtual bool IsPrefix(string source, string prefix, CompareOptions options)` doesn't call any virtual methods. And unless someone explicitly casts the string to Span, the following method `public virtual bool IsPrefix(string source, string prefix)` will not call the internal IsPrefix(ROS...) method.

Given that no public surface area is changing, and the public virtual methods that have new span overloads are not calling other virtual methods, I am not seeing the potential issues.

Reply:
CompareInfo does not have a public instance constructor, so it is de-facto sealed. So this is not a problem.